### ISSUE TITLE
cannot upgrade state, missing MigrateState function

### ISSUE THREAD

**Issue by syl20bnr** [terraform-providers/terraform-provider-google#3718]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment
* If an issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to "hashibot", a community member has claimed the issue already.

### Terraform Version

0.12

### Affected Resource(s)

Not sure. Here is the state with the list of resources:

```
data.terraform_remote_state.admin
data.terraform_remote_state.xpn_host
aws_route53_record.eta_lb
google_project_service.compute
module.eta.google_sql_database.eta_stag
module.eta.google_sql_user.api
module.eta.module.cron.data.google_compute_instance.instances
module.eta.module.cron.data.google_compute_region_instance_group.details
module.eta.module.cron.google_compute_instance_template.template
module.eta.module.cron.google_compute_region_instance_group_manager.http
module.eta.module.cron.google_project_service.compute
module.eta.module.front.data.google_compute_instance.instances[0]
module.eta.module.front.data.google_compute_instance.instances[1]
module.eta.module.front.data.google_compute_region_instance_group.details
module.eta.module.front.google_compute_instance_template.template
module.eta.module.front.google_compute_region_instance_group_manager.http
module.eta.module.front.google_project_service.compute
module.eta.module.front_lb.google_compute_address.ip
module.eta.module.front_lb.google_compute_forwarding_rule.https
module.eta.module.front_lb.google_compute_health_check.https
module.eta.module.front_lb.google_compute_region_backend_service.https
module.eta.module.psql.aws_route53_record.master_endpoint
module.eta.module.psql.aws_route53_record.master_endpoint_public
module.eta.module.psql.aws_route53_record.master_shortcut
module.eta.module.psql.google_project_service.network
module.eta.module.psql.google_project_service.network_xpn_project
module.eta.module.psql.google_sql_database.csdba
module.eta.module.psql.google_sql_database_instance.master
module.eta.module.psql.google_sql_user.csdba
module.eta.module.psql.random_id.master_suffix
module.eta.module.redis.aws_route53_record.endpoint
module.eta.module.redis.google_project_service.memorystore_redis
module.eta.module.redis.google_redis_instance.redis_instance
module.eta.module.redis.random_id.suffix
module.eta.module.front_lb.module.authorized_cidrs_firewall_rule.google_compute_firewall.allow_ports
module.eta.module.front_lb.module.health_check_firewall_rule.google_compute_firewall.allow_ports
```

### Debug Output

The end of the TRACE:

```
2019/05/24 16:16:41 [TRACE] module.eta.module.psql: eval: *terraform.EvalWriteState
2019/05/24 16:16:41 [TRACE] EvalWriteState: writing current state object for module.eta.module.psql.google_sql_database.csdba
2019/05/24 16:16:41 [TRACE] [walkRefresh] Exiting eval tree: module.eta.module.psql.google_sql_database.csdba
2019/05/24 16:16:41 [TRACE] vertex "module.eta.module.psql.google_sql_database.csdba": visit complete
2019/05/24 16:16:41 [TRACE] vertex "module.eta.module.psql.google_sql_database.csdba": dynamic subgraph completed successfully
2019/05/24 16:16:41 [TRACE] vertex "module.eta.module.psql.google_sql_database.csdba": visit complete
2019/05/24 16:16:41 [TRACE] dag/walk: upstream of "provider.google (close)" errored, so skipping
2019/05/24 16:16:41 [TRACE] dag/walk: upstream of "root" errored, so skipping

Error: cannot upgrade state, missing MigrateState function
```

### Steps to Reproduce

Happens during a plan.

**Comment by syl20bnr** [comment]
I migrated other states to 0.12 successfully, this one is the first state with an internal LB so it might be a clue.

**Comment by syl20bnr** [comment]
The error could come from the resource `compute_region_backend_service` as it is the only resource we don't find in our other states.

**Comment by steven-edgar** [comment]
I get this same issue with one of our projects. I've traced the issue back to google_compute_region_backend_service, or possibly google_compute_forwarding_rule. I traced this down by running a "terraform plan -target=...." on each of the resources in my terraform file, and it was only those two which generated this error. Our forwarding rule references the backend service, and has a depends_on entry for it, so as OP noted in comments I'm reasonably sure the actual problem lies in google_compute_region_backend_service, as a plan on either of those resources will result in backend_service module being called. 

Removing these two resources from our terraform does not remove the error when running "terraform plan" without any target specified, presumably because they still exist in our state file and therefore the module is still invoked. 

Terraform 0.12
google provider 2.7
